src/aiida/orm/authinfos.py

```diff
@@ -134,10 +135,22 @@ def get_auth_params(self) -> Dict[str, Any]:
         return self._backend_entity.get_auth_params()
 
     def set_auth_params(self, auth_params: Dict[str, Any]) -> None:
-        """Set the dictionary of authentication parameters
+        """Set the dictionary of authentication parameters.
+
+        If password present in `auth_params`, it stores it in secure storage and obfuscates it.
 
         :param auth_params: a dictionary with authentication parameters
         """
+        import copy
+
+        auth_params = copy.deepcopy(auth_params)
+        # the default value for the password in CLI is the empty string
+        # which is covered by evaluating to False in the conditional statement
```

Comment:
Note that this prevents usage of empty password over CLI. This seems to be a use case of some people looking at the paramiko, but I don't think it is possible with our CLI because you cannot distinguish an empty string and None entry if you want to give prompts for each parameter

Reply:
This is not clear to me. There are two ways to set answer to CLI prompt without give explicit input, either by press "enter" to continue, which use the default value, or by type `!` to skip the set which I think will set it to `None`. What is the case for the auth_params? I believe for this password which is the password to the remote can never by a empty. So it would make sense to not have default value when press "enter", when user type `!` it should raise.